gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMethodDescriptor.java

```diff
@@ -30,8 +30,115 @@
 package com.google.api.gax.httpjson;
 
 import com.google.api.core.BetaApi;
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.Set;
 
 @BetaApi
-public class ApiMethodDescriptor<RequestT, ResponseT> {
-  // TODO implement
+@AutoValue
+/* Method descriptor for messages to be transmitted over HTTP. */
+public abstract class ApiMethodDescriptor<RequestT, ResponseT> {
+  public abstract String fullMethodName();
+
+  public abstract Gson baseGson();
+
+  public abstract Gson requestMarshaller();
+
+  public abstract Gson responseMarshaller();
+
+  public abstract Type requestType();
+
+  public abstract Type responseType();
+
+  public abstract Set<String> pathParams();
+
+  public abstract Set<String> queryParams();
+
+  public abstract HttpMethod httpMethod();
+
+  public abstract HttpRequestFormatter httpRequestBuilder();
+
+  /* In the form "[prefix]%s[suffix]", where
+   *    [prefix] is any string; if length greater than 0, it should end with '/'.
+   *    [suffix] is any string; if length greater than 0, it should begin with '/'.
+   * This String format is applied to a serialized ResourceName to create the relative endpoint path.
+   */
+  public abstract String endpointPathTemplate();
```

Comment:
use the `getX()` form for getter methods in this class

Reply:
Done.